src/scheduler.c

```diff
@@ -359,15 +364,68 @@ static int may_sleep(jl_ptls_t ptls) JL_NOTSAFEPOINT
 }
 
 
+// jl_n_threads_spinning is basically the same as n_threads_running with the caveat that it gets decremented when a thread starts doing work
+// if a thread leaves to do work and spinning goes to 0 we try to wake up another thread
+STATIC_INLINE void add_spinner(jl_task_t *ct) JL_NOTSAFEPOINT
```

Comment:
Could you add some comments around the extra synchronization added here? It isn't entirely clear why we would want this many added atomics in the fast path

Reply:
The synchronization we have here is copied from wakeup_thread. Which this code is replacing. It does add another fetch_add which I'm not super happy about to be clear, specially since n_threads_running is almost exactly the same counter so I probably want to rework this a bit.